intellij/org.eclipse.xtend.idea/src/org/eclipse/xtend/core/idea/highlighting/XtendHighlightingLexer.xtend

```diff
@@ -99,10 +100,7 @@ class XtendHighlightingLexer extends LexerBase {
 	}
 
 	override getState() {
-		if (currentRichTextToken != null)
-			currentRichTextToken.type
-		else
-			delegate.state
+		return if (_currentRichTextToken === null) 0 else 1
```

Comment:
No, it's never called with something other than the base state (0) and we couldn't set the parser into that other state, as we don't know the rich string token.